server/src/main/java/org/elasticsearch/cluster/routing/ShardMovementWriteLoadSimulator.java

```diff
@@ -87,9 +92,46 @@ public Map<String, NodeUsageStatsForThreadPools> simulatedNodeUsageStatsForThrea
                 adjustedNodeUsageStatsForThreadPools.put(entry.getKey(), entry.getValue());
             }
         }
+
+        // Add `NodeUsageStatsForThreadPools` for any nodes not present in the original `NodeUsageStatsForThreadPools` map.
+        addUsageStatsForAnyNodesNotPresentInOriginalNodeUsageStatsForThreadPools(adjustedNodeUsageStatsForThreadPools);
+
         return Collections.unmodifiableMap(adjustedNodeUsageStatsForThreadPools);
     }
 
+    private void addUsageStatsForAnyNodesNotPresentInOriginalNodeUsageStatsForThreadPools(
+        Map<String, NodeUsageStatsForThreadPools> nodeUsageStatsForThreadPools
+    ) {
+        // Assume the new node has the same size thread pool as the largest existing node
+        final OptionalInt largestWriteThreadPool = originalNodeUsageStatsForThreadPools.values()
+            .stream()
+            .map(NodeUsageStatsForThreadPools::threadPoolUsageStatsMap)
+            .map(m -> m.get(ThreadPool.Names.WRITE))
+            .mapToInt(NodeUsageStatsForThreadPools.ThreadPoolUsageStats::totalThreadPoolThreads)
+            .max();
+
+        if (largestWriteThreadPool.isPresent()) {
+            simulatedNodeWriteLoadDeltas.forEach((ObjectDoubleProcedure<? super String>) (nodeId, writeLoadDelta) -> {
+                nodeUsageStatsForThreadPools.computeIfAbsent(
+                    nodeId,
+                    missingNodeId -> new NodeUsageStatsForThreadPools(
+                        missingNodeId,
+                        Map.of(
+                            ThreadPool.Names.WRITE,
+                            new NodeUsageStatsForThreadPools.ThreadPoolUsageStats(
+                                largestWriteThreadPool.getAsInt(),
+                                updateNodeUtilizationWithShardMovements(0.0f, (float) writeLoadDelta, largestWriteThreadPool.getAsInt()),
+                                0
+                            )
+                        )
+                    )
+                );
+            });
+        } else {
+            logger.debug("No nodes found to estimate write thread pool size, skipping");
```

Comment:
If we get here, there were no other nodes in the `ClusterInfo` to base our estimate off of. In this case we could also assume the same pool size as the local node perhaps? but we don't have that information in here. This also seems very unlikely to actually occur.